Feature that enable modify hydra configuration group through command line

ooctipus opened the pull request:
# Description
This PR introduce a new feature in interacting with hydra, bring group override feature to isaac lab

Demo Script
![image](https://github.com/user-attachments/assets/56e1c991-ed86-47fc-97aa-6463d567700a)

## Demo 1: Overriding Event Managers with Pre-grouped Alternative Managers

### Example 1: Overriding Event Manager

```bash
# Default
python train.py <default>


Output:
Event Terms: ["reset_robot_joint_position"]
```

```bash
# Override to include friction
python train.py env.events=rand_joint_pos_friction



Output:
Event Terms: ["reset_robot_joints", "reset_robot_joint_friction"]
```

```bash
# Override to include armature as well
python train.py env.events=rand_joint_pos_friction_amarture



Output:
Event Terms: ["reset_robot_joints", "reset_robot_joint_friction", "reset_robot_joint_amature"]
```

---

### Example 2: Overriding Observation Manager

```bash
# Default
python train.py <default>



Output:
joint_pos.noise.n_min   -> -0.01
joint_pos.noise.n_max   ->  0.01
enable_corruption       ->  True
```

```bash
# Disable noise and corruption
python train.py env.observations=state_obs_no_noise



Output:
joint_pos.noise         -> None
enable_corruption       -> False
```

---

## Demo 2: Overriding Terms Under a Manager

### Example: Change Only `arm_action` Term Under `ActionManager`

```bash
# Default action manager
python train.py <default>



Output:
env.actions.arm_action  -> JointPositionActionCfg
```

```bash
# Switch to inverse-kinematics action
python train.py env.actions.arm_action=ik_abs_arm_action



Output:
env.actions.arm_action  -> DifferentialInverseKinematicsActionCfg
```

```bash
# Switch to operational-space control
python train.py env.actions.arm_action=osc_arm_action



Output:
env.actions.arm_action  -> OperationalSpaceControllerActionCfg
```

---

## Demo 3: Overriding `agent.policy` Configuration

Demo Script:
![image](https://github.com/user-attachments/assets/0519116e-7880-4f4a-bba4-935c91322171)


```bash
# Default policy network
python.train.py <default>



Output:
Actor MLP: Sequential(
  (0): Linear(in_features=32, out_features=64, bias=True)
  (2): Linear(in_features=64, out_features=64, bias=True)
  (4): Linear(in_features=64, out_features=7, bias=True)
)
```

```bash
# Use a medium-sized network
python train.py agent.policy=medium_network



Output:
Actor MLP: Sequential(
  (0): Linear(in_features=32, out_features=256, bias=True)
  (2): Linear(in_features=256, out_features=128, bias=True)
  (4): Linear(in_features=128, out_features=64, bias=True)
  (6): Linear(in_features=64, out_features=7, bias=True)
)
```  
Fixes # (issue)

<!-- As a practice, it is recommended to open an issue to have discussions on the proposed pull request.
This makes it easier for the community to keep track of what is being developed or added, and if a given feature
is demanded by more than one party. -->

## Type of change

- New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./isaaclab.sh --format`
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the changelog and the corresponding version in the extension's `config/extension.toml` file
- [x] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

<!--
As you go through the checklist above, you can mark something as done by putting an x character in it

For example,
- [x] I have done this task
- [ ] I have not done this task
-->

kellyguo11 commented on source/isaaclab_tasks/isaaclab_tasks/utils/hydra.py:
```diff
@@ -55,9 +57,31 @@ def register_task_to_hydra(
+    config_store = ConfigStore.instance()
+
+    has_hydra_group_configuration = "configurable_entry_point" in gym.spec(task_name).kwargs
+    if has_hydra_group_configuration:
```
is there a good way to test this in the unit tests?

ooctipus commented:
Maybe we should move the all of them to Hydra Group Configurables

instead of 
```
gym.register(
    id="Isaac-Cartpole-RGB-v0",
    entry_point="isaaclab.envs:ManagerBasedRLEnv",
    disable_env_checker=True,
    kwargs={
        "env_cfg_entry_point": f"{__name__}.cartpole_camera_env_cfg:CartpoleRGBCameraEnvCfg",
        "rl_games_cfg_entry_point": f"{agents.__name__}:rl_games_camera_ppo_cfg.yaml",
    },
)

gym.register(
    id="Isaac-Cartpole-RGB-ResNet18-v0",
    entry_point="isaaclab.envs:ManagerBasedRLEnv",
    disable_env_checker=True,
    kwargs={
        "env_cfg_entry_point": f"{__name__}.cartpole_camera_env_cfg:CartpoleResNet18CameraEnvCfg",
        "rl_games_cfg_entry_point": f"{agents.__name__}:rl_games_feature_ppo_cfg.yaml",
    },
)

gym.register(
    id="Isaac-Cartpole-RGB-TheiaTiny-v0",
    entry_point="isaaclab.envs:ManagerBasedRLEnv",
    disable_env_checker=True,
    kwargs={
        "env_cfg_entry_point": f"{__name__}.cartpole_camera_env_cfg:CartpoleTheiaTinyCameraEnvCfg",
        "rl_games_cfg_entry_point": f"{agents.__name__}:rl_games_feature_ppo_cfg.yaml",
    },
)
```


do:
```
gym.register(
    id="Isaac-Cartpole-RGB-v0",
    entry_point="isaaclab.envs:ManagerBasedRLEnv",
    disable_env_checker=True,
    kwargs={
        "env_cfg_entry_point": f"{__name__}.cartpole_camera_env_cfg:CartpoleRGBCameraEnvCfg",
        "rl_games_cfg_entry_point": f"{agents.__name__}:rl_games_camera_ppo_cfg.yaml",
        "configurable_entry_point": f"{agents.__name__}.configurables:AgentConfigurables",
    },
)
```

`python train.py --task Isaac-Cartpole-RGB-v0 --encoder ResNet18`
`python train.py --task Isaac-Cartpole-RGB-v0 --encoder TheiaTiny`

ooctipus commented:
I have added tests to verify my feature is effective and other environments that doesn't have the feature does not break, and added the documentation section under Features/Hydra Configuration System. @kellyguo11 Ready for Viz
